salt/utils/jinja.py

```diff
@@ -707,7 +707,7 @@ def method_call(obj, f_name, *f_args, **f_kwargs):
     return getattr(obj, f_name, lambda *args, **kwargs: None)(*f_args, **f_kwargs)
 
 
-@jinja2.contextfunction
+@jinja2.pass_context
```

Comment:
The `pass_context` function was added in 3.0.0 here https://github.com/pallets/jinja/pull/1389 .

If we update this function call it will fail for users still on jinja <3.0.0. Can we detect the jinja versions and then use the correct call?

Also this will require a changelog.

Reply:
That would be great! 
> Can we detect the jinja versions and then use the correct call?

I don’t know how to implement this, any sample code?